### PR TITLE
Adding progressBorderColor property on progressBar

### PR DESCRIPTION
**Pull request by bardez**
Used to allow percent indicator uses a different border color

Screenshot 2023-05-12 at 16 55 34

Screenshot 2023-05-12 at 16 55 46

### CODE REVIEW

File: lib/circular_percent_indicator.dart [hunk 1]
```diff
@@ -408,9 +416,16 @@ class _CirclePainter extends CustomPainter {
     }
 
     _paintLine.color = progressColor;
-    _paintLine.style = PaintingStyle.stroke;
-    _paintLine.strokeWidth = lineWidth;
+    _paintLine.style = PaintingStyle.fill;
+    _paintLine.strokeWidth = lineWidth - 2;
```
Comment (diegoveloper) [inline lib/circular_percent_indicator.dart:420]: why -2 here?

**Comment by diegoveloper** [comment]
thanks for making these changes, can you also update the examples, adding a few using this new property? thanks

File: lib/circular_percent_indicator.dart (hunk 1 above)
Comment (bardez) [reply lib/circular_percent_indicator.dart:420]: Just for border size ... I think if it is dynamic, another problems will appear

**Comment by bardez** [comment]
I make some changes to avoid percent border on linear. please verify and accept. Now with some examples

**Comment by bardez** [comment]
Screenshot 2023-05-17 at 11 16 06

Screenshot 2023-05-17 at 11 17 06

File: lib/linear_percent_indicator.dart [hunk 2]
```diff
@@ -346,6 +354,12 @@ class _LinearPainter extends CustomPainter {
     _paintLine.color = progress.toString() == "0.0"
         ? progressColor.withOpacity(0.0)
         : progressColor;
+
+    if (progressBorderColor != null) {
+      _paintLineBorder.color = progress.toString() == "0.0"
```
Comment (diegoveloper) [inline lib/linear_percent_indicator.dart:359]: can you reuse/refactor this condition?  `progress.toString() == "0.0"`

Comment (bardez) [reply lib/linear_percent_indicator.dart:359]: I just copy from line 354, but I'll refactor

File: lib/linear_percent_indicator.dart [hunk 3]
```diff
@@ -381,10 +403,21 @@ class _LinearPainter extends CustomPainter {
...
         _paintLine.shader = _createGradientShaderLeftToRight(size, xProgress);
       }
+      if (progressBorderColor != null) {
+        linePathBorder.addRRect(RRect.fromRectAndRadius(
+            Rect.fromLTWH(0, 0, xProgress, size.height), barRadius));
+      }
       linePath.addRRect(RRect.fromRectAndRadius(
-          Rect.fromLTWH(0, 0, xProgress, size.height), barRadius));
+          Rect.fromLTWH(
+              factor, factor, xProgress - factor * 2, size.height - factor * 2),
```
Comment (diegoveloper) [inline lib/linear_percent_indicator.dart:416]: Also here, extract the factor *2

**Comment by bardez** [comment]
Up

**Comment by bardez** [comment]
thanks bro!

**Pull request merged**